reverse_engineering/helpers/connectionHelper.js

```diff
@@ -116,10 +123,40 @@ const createClient = async (connectionInfo, logger) => {
 		application_name: 'Hackolade',
 	};
 
+	const client = await connectClient(config).catch(retryOnSslError(connectionInfo, config, logger));
+
+	return { client, sshTunnel };
+};
+
+const retryOnSslError = (connectionInfo, config, logger) => async error => {
+	if (error.message === SSL_NOT_SUPPORTED_MESSAGE && connectionInfo.sslType === 'prefer') {
+		logger.info("Retry connection without SSL (SSL mode 'prefer')");
+		logger.error(error);
+
+		return await connectClient({
+			...config,
+			ssl: false,
+		});
+	}
+
+	if (error.code?.toString() === POSTGRES_SSL_REQUIRED_ERROR_CODE && connectionInfo.sslType === 'allow') {
+		logger.info("Retry connection with SSL (SSL mode 'allow')");
+		logger.error(error);
+
+		return await connectClient({
+			...config,
+			ssl: { rejectUnauthorized: false },
```

Comment:
@Vitalii4as hey Vitalii, do you think falling back silently to ignoring self signed certificates is a good idea? Shouldn't we support that as an option and ask the user to retry after he setup it?

Reply:
I implemented the logic described in the docs https://www.postgresql.org/docs/14/libpq-ssl.html#LIBPQ-SSL-SSLMODE-STATEMENTS. For sslmode=allow we try to connect without SSL firstly but if server insist on it we connect with SSL. It works the same as pgAdmin and psql. Certificates are verified only with sslmode 'verify-ca' and 'verify-full'.